src/rust/engine/pantsd/src/lib.rs

```diff
@@ -35,11 +34,35 @@ use std::{fmt, fs};
 
 use libc::pid_t;
 use log::debug;
-use options::{option_id, BuildRoot, OptionId, OptionType};
+use options::{option_id, BuildRoot, OptionId, OptionParser, OptionType};
 use sha2::digest::Update;
 use sha2::{Digest, Sha256};
 use sysinfo::{ProcessExt, ProcessStatus, System, SystemExt};
 
+pub struct ConnectionSettings {
+  pub port: u16,
+  pub timeout_limit: f64,
+  pub dynamic_ui: bool,
+}
+
+impl ConnectionSettings {
+  pub fn new(port: u16) -> ConnectionSettings {
+    ConnectionSettings {
+      port,
+      timeout_limit: 60.0,
+      dynamic_ui: true,
+    }
+  }
+}
+
+fn to_hex(bytes: &[u8]) -> String {
```

Comment:
I note that the workspace already depends on https://docs.rs/hex/0.4.3/hex/, so callers of this could probably be `hex::encode(...)`.

(I imagine it'll also be more efficient by side-stepping the format machinery/indirection, but that's presumably not going to matter here at all.)

Reply:
Yea, good call. Will use it here.

Digging back to ancient history, I'm not sure why we didn't use the `hex` crate for producing fingerprints there either: https://github.com/pantsbuild/pants/blame/8097749a3268b5f3e302f388a8df327539de1904/src/rust/engine/hashing/src/lib.rs#L86-L93 ... but I'll add a TODO and avoid doing that in this change.